spec/models/mou_signature_spec.rb

```diff
@@ -0,0 +1,69 @@
+require "rails_helper"
+
+RSpec.describe MouSignature, type: :model do
+  let(:organisation) { create(:organisation) }
+  let(:user) { create(:user, organisation:) }
+  let(:mou_signature) do
+    described_class.create!(
+      user:,
+      organisation:,
+      agreed_at: Time.zone.now,
+    )
+  end
+
+  it "setting user to nil triggers a DB constraint" do
+    mou_signature.user = nil
+    expect { mou_signature.save(validate: false) }.to raise_error(ActiveRecord::StatementInvalid).with_message(/null value in column "user_id"/)
+  end
```

Comment:
Could this not be a model validation to catch it before it gets to the DB?

Reply:
I've gone for DB constraints over model validations as they keep the DB in order and seem a bit more solid. I'm also reading https://sustainable-rails.com/ which recommends them over model validations :)

Reply:
sure DB constraints are good but its better for Rails to handle those errors before it gets there. I would say its in addition rather than instead of DB. Also we have control over those errorr,
 if we migrate away from postgres the new db or Postgres changes the error message and returns a different error message meaning this test fails through no fault of ours.
 
 Could use  `belongs_to :user` in the model.

Reply:
Keep DB constraints but get rid of the tests and rely on belongs_to  in the model and have tests to make sure the model is invalid if a user is nil and you could check what error message is returned.

Reply:
got rid of em 👍